src/main/java/com/bookstore/service/CategoryServiceImpl.java

```diff
@@ -0,0 +1,87 @@
+package com.bookstore.service;
+
+import com.bookstore.dto.CategoryDto;
+import com.bookstore.dto.CategoryRequestDto;
+import com.bookstore.entity.Category;
+import com.bookstore.exception.EntityAlreadyExistsException;
+import com.bookstore.exception.EntityNotFoundException;
+import com.bookstore.mapper.CategoryMapper;
+import com.bookstore.repository.CategoryRepository;
+import java.util.List;
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+    private final LocaleService localeService;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable).stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        Category category = getModelById(id);
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public CategoryDto save(CategoryRequestDto categoryDto) {
+        if (categoryRepository.existsByName(categoryDto.getName())) {
+            throw getAlreadyExistsException();
+        }
+        Category category = categoryMapper.toModel(categoryDto);
+        Category savedCategory = categoryRepository.save(category);
+        return categoryMapper.toDto(savedCategory);
+    }
+
+    @Override
+    public CategoryDto update(Long id, CategoryRequestDto categoryDto) {
+        Category categoryFromDB = getModelById(id);
+        String newName = categoryDto.getName();
+
+        if (!Objects.equals(categoryFromDB.getName(), newName)
+                && categoryRepository.existsByName(newName)) {
```

Comment:
a little bit confusing with this checking, if the category already exists by name you throw an exception. What for?